sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -210,9 +219,9 @@ public void describeTo(Description description) {
   @Override
   public void describeMismatchSafely(PipelineResult pResult, Description description) {
     String info;
-    if (response == null || response.getRows() == null || response.getRows().isEmpty()) {
-      // invalid query response
-      info = String.format("Invalid BigQuery response: %s", Objects.toString(response));
+    if (!response.getJobComplete()) {
+      // query job not complete
+      info = String.format("Query job not complete with retry. Got response: %s", response);
```

Comment:
query response contains jobID. `response.toString()` is in Json format so I think it's pretty readable. Previously, response can be null, so `Objects.toString(null)` can print "null" and avoid NullPointerException.